models/service.rb

```diff
@@ -58,5 +58,14 @@ class Service < Base
     has_many :sticky_vehicles, class_name: 'Models::Vehicle'
     has_many :quantities, class_name: 'Models::Quantity'
     has_many :relations, class_name: 'Models::Relation'
+
+    def route_activity(index)
+      Models::RouteActivity.new(
+        service_id: self.original_id || self.id,
```

Comment:
maybe this attribute could be only "id". Since you defined rests as an activity I think it is better to have the most neutral field.